Scheduling options cannot be null.

Hi!

I am trying to create several datadog monitors at once by looping through a map. Not all monitors in the map are the same and some will have settings that are not present in the others. To that end I have been including the settings like so:
```
  monitor_threshold_windows {
    recovery_window = try(each.value.monitor_threshold_windows[0].recovery_window, null)
    trigger_window  = try(each.value.monitor_threshold_windows[0].trigger_window, null)
  }
```
Which has generally been working fine. However when I try and do similar for the Scheduling options block
```
  scheduling_options {
    evaluation_window {
      day_starts = try(each.value.day_starts, null)
      hour_starts = try(each.value.hour_starts, null)
      month_starts = try(each.value.month_starts, null)
    }
  }
```
The plugin crashes. I initially thought this might be an issue with looping, but even just having the block present like this
```
  scheduling_options {
    evaluation_window {
    }
  }
```
causes the same crash. I imagine this is just a difference in the monitor_threshold_windows supporting null values and the scheduling_options not.
### Terraform Version

Terraform v1.3.6
on darwin_arm64
+ provider registry.terraform.io/datadog/datadog v3.23.0

### Affected Resource(s)

datadog_monitor
### plugin crash output
```
Stack trace from the terraform-provider-datadog_v3.23.0 plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}
```
https://gist.github.com/zexoor/f7a5fc63185c9dac66a8b7db9a196341

--- a/datadog/resource_datadog_monitor.go
+++ b/datadog/resource_datadog_monitor.go
@@ -655,24 +655,26 @@ func buildMonitorStruct(d utils.Resource) (*datadogV1.Monitor, *datadogV1.Monito
 
 	if attr, ok := d.GetOk("scheduling_options"); ok {
 		scheduling_options_list := attr.([]interface{})
-		scheduling_options_map := scheduling_options_list[0].(map[string]interface{})
-		evaluation_window_map := scheduling_options_map["evaluation_window"].([]interface{})[0].(map[string]interface{})
-		scheduling_options := datadogV1.NewMonitorOptionsSchedulingOptions()
-		evaluation_window := datadogV1.NewMonitorOptionsSchedulingOptionsEvaluationWindow()
-		day_month_scheduling := false
-		if day_starts, ok := evaluation_window_map["day_starts"].(string); ok && day_starts != "" {
-			evaluation_window.SetDayStarts(day_starts)
-			day_month_scheduling = true
+		if scheduling_options_map, ok := scheduling_options_list[0].(map[string]interface{}); ok {
+			if evaluation_window_map, ok := scheduling_options_map["evaluation_window"].([]interface{})[0].(map[string]interface{}); ok {
+				scheduling_options := datadogV1.NewMonitorOptionsSchedulingOptions()
+				evaluation_window := datadogV1.NewMonitorOptionsSchedulingOptionsEvaluationWindow()
+				day_month_scheduling := false
+				if day_starts, ok := evaluation_window_map["day_starts"].(string); ok && day_starts != "" {
+					evaluation_window.SetDayStarts(day_starts)
+					day_month_scheduling = true
+				}
+				if month_starts, ok := evaluation_window_map["month_starts"].(int); ok && month_starts != 0 {
+					evaluation_window.SetMonthStarts(int32(month_starts))
+					day_month_scheduling = true
+				}
+				if hour_starts, ok := evaluation_window_map["hour_starts"].(int); ok && !day_month_scheduling {
+					evaluation_window.SetHourStarts(int32(hour_starts))
+				}
+				scheduling_options.SetEvaluationWindow(*evaluation_window)
+				o.SetSchedulingOptions(*scheduling_options)
+			}
 		}
-		if month_starts, ok := evaluation_window_map["month_starts"].(int); ok && month_starts != 0 {
-			evaluation_window.SetMonthStarts(int32(month_starts))
-			day_month_scheduling = true
-		}
-		if hour_starts, ok := evaluation_window_map["hour_starts"].(int); ok && !day_month_scheduling {
-			evaluation_window.SetHourStarts(int32(hour_starts))
-		}
-		scheduling_options.SetEvaluationWindow(*evaluation_window)
-		o.SetSchedulingOptions(*scheduling_options)
 	}
 
 	if attr, ok := d.GetOk("notification_preset_name"); ok {

